Add tooltips to display name and email

onehappycat opened the pull request:
#18158 Add tooltips to ~~username~~ 'display name' and 'email' fields in user management.

kesselb commented:
Thanks :+1: 

Would you mind to share a screenshot? ;)

onehappycat commented:
> Thanks 👍
> 
> Would you mind to share a screenshot? ;)

<img width="466" alt="Screenshot 2019-12-02 at 19 45 08" src="https://user-images.githubusercontent.com/56887614/69985864-5fefe600-153c-11ea-8ff5-84784047e157.png">

skjnldsv reviewed (changes requested):
Hum, this will conflict with @GretaD's work on the new layout #17239 
I would rather have this wait until the other to ease the potential rebase :)

GretaD commented:
yes please lets wait, not that much left to do on my ticket. I'll ping you again again.

onehappycat commented:
Sounds good. The new layout looks amazing. 👍

blizzz commented:
@GretaD's PR was merged. What to do with this one?

skjnldsv commented:
rework it to add the tooltips at the correct place :)

onehappycat committed:
Add tooltips to display name and email fields in Users

Signed-off-by: onehappycat <onehappycat@gmx.com>

onehappycat commented:
Rebased to current master, added the tooltips to correct places at the new layout.

Before:
<img width="786" alt="Screenshot 2019-12-14 at 14 31 28" src="https://user-images.githubusercontent.com/56887614/70849428-76703880-1e7e-11ea-9bd5-de4df0a7f5a3.png">

After:
<img width="759" alt="Screenshot 2019-12-14 at 14 31 36" src="https://user-images.githubusercontent.com/56887614/70849430-7839fc00-1e7e-11ea-85dc-a3f80611425f.png">

skjnldsv commented on apps/settings/src/components/UserList/UserRowSimple.vue:
```diff
@@ -15,12 +15,18 @@
 			{{ user.id }}
 			<div class="displayName subtitle">
-				{{ user.displayname }}
+				<div v-if="user.displayname.length>20" v-tooltip.auto="user.displayname">
```
Hum, you could also just ellipsis with css and enable the tooltip only when the length is bigger than '20 char' ?

skjnldsv replied:
Also, we're using the three dot ellipsis character, not three real dots: `…`
And with a space before :)

skjnldsv reviewed (changes requested):
See comments ;)

onehappycat commented:
Rebased and added the full output of `make build-js-production`, but webpack is still complaining.

skjnldsv commented:
/compile amend /

skjnldsv commented:
@onehappycat something else was merged, please rebase again :p
After rebase you can use the `/compile amend /` command to trigger a compile of any js file by our bot ;)

ChristophWurst commented on apps/settings/src/components/UserList/UserRowSimple.vue:
```diff
@@ -15,12 +15,16 @@
 			{{ user.id }}
 			<div class="displayName subtitle">
-				{{ user.displayname }}
+				<div v-tooltip="user.displayname.length > 20 ? user.displayname : ''" class="cellText">
```
is there any other way to detect the truncated text? the magic number 20 doesn't care about different fonts, letter width and dynamic size of the conainer.

skjnldsv replied:
Yes, use a vue ellipsis component/directive :)
In the meantime, I'd say let get this in, worst thing that could happen is a tooltip :see_no_evil:

ChristophWurst reviewed (approved):
I like happy cats
